src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -133,12 +133,21 @@ variant_add_commit_details (GVariantDict *dict,
 {
   g_autoptr(GVariant) metadata = NULL;
   g_autofree gchar *version_commit = NULL;
+  g_autofree gchar *end_of_life = NULL;
   guint64 timestamp = 0;
 
   timestamp = ostree_commit_get_timestamp (commit);
   metadata = g_variant_get_child_value (commit, 0);
   if (metadata != NULL)
-    g_variant_lookup (metadata, "version", "s", &version_commit);
+    {
+      g_variant_lookup (metadata, "version", "s", &version_commit);
+      /* avoid double insertion */
+      if (prefix != NULL)
+        g_variant_lookup(metadata, "end-of-life", "s", &end_of_life);
```

Comment:
Done in https://github.com/ostreedev/ostree/pull/1024

I also noted that we *also* have `ostree.endoflife` mentioned in the test suite; I think for consistency we should change the bits above to be `ostree.endoflife` since we already support `ostree.endoflife-rebase`.  Want to review my PR and once it lands, do a patch on top of it?